src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -6,17 +6,81 @@ description: Motion and how it works in the NUbots codebase.
 slug: /system/subsystems/motion
 ---
 
-The motion subsystem contains modules that are responsible for walking, kicking, moving the head, and running scripts. They perform these motions by sending servo commands (called targets) to the motors.
+This page details motion modules within the NUbots main codebase, such as the walk engines and the Script Engine. The Script Engine is detailed on this page, while Script Runner and Script Tuner can be found on the [Behaviour](/system/subsystems/behaviour#script-runner) page.
+
+
+## Walk Engine
+
+Currently NUbots has two walk engines. One is an **open loop hybrid walk controller** and the other is an **open loop walk** based on **quintic splines**. These engines take care of the mathematics and control involved in taking steps without falling over. The walk engines do not plan where to walk to, only how to walk.
+
+A walk engine will trigger when a [**walkcommand**](https://github.com/NUbots/NUbots/blob/master/shared/message/motion/WalkCommand.proto) is [emitted](/system/foundations/nuclear#emit-statements). The walk command gives a vector [x, y, z] which represents:
+
+| Variable | Description                                                                                |
+| -------- | ------------------------------------------------------------------------------------------ |
+| x        | velocity in metres per second                                                              |
+| y        | velocity in metres per second                                                              |
+| z        | yaw in radians per second, positive is counter-clockwise. The 0 point is straight forward. |
+
+The [behaviour](/system/subsystems/behaviour) of the robot determines what the walk command will be.
+
+### Open Loop Hybrid Walk
+
+This walk is an **open loop hybrid walk controller**, dynamically switching between an analytically solved trajectory for the centre of mass based on the **linear inverted pendulum model**, and a **Zero Moment Point preview controller**. To add local feedback, positions for the pitch and roll ankle joints are prescribed based on information from the Inertial Measurement Unit (IMU) to counteract small imbalances.
```

Comment:
are there good resources for learning about inverted pendulum models and zmp controllers? could be worth linking for further reading here - unless it's not worth it because we're moving away from this walk anyway

Reply:
This might as well be removed since we're removing it from the codebase now.

Reply:
:grin: